components/exercises/chapter-5-oscar/YourOwnLoop/loop.test.tsx

```diff
@@ -0,0 +1,48 @@
+import { loop } from './loop';
+
+describe('loop function', () => {
+  // Mock the setInterval function to control time in tests
+  jest.useFakeTimers();
+
+  it('should execute body function until test function returns false', () => {
+    let value = 0;
+
+    // Define the test function
+    const testFunction = (value: number) => value < 5;
+
+    // Define the update function
+    const updateFunction = (value: number) => value + 1;
+
+    // Define a mock body function to track calls
+    const mockBody = jest.fn();
+
+    // Call the loop function
+    loop(value, testFunction, updateFunction, mockBody);
+
+    // Fast forward timers to allow for multiple iterations
+    jest.advanceTimersByTime(5000);
+
+    // Expect that the mock body function was called 5 times (as per the test condition)
+    expect(mockBody).toHaveBeenCalledTimes(5);
+  });
```

Comment:
When you update the function to remove `setInterval` you should be able to test soemthing way simpler, like

```js
const mockBody = jest.fn();

loop(0, (value) => value < 3, (value) => value + 1, mockBody)

expect(mockBody).toHaveBeenCalledTimes(3);
expect(mockBody).toHaveBeenCalledWith(0);
expect(mockBody).toHaveBeenCalledWith(1);
expect(mockBody).toHaveBeenCalledWith(2);
```

Reply:
refactored, thank you